Bug: Query method doesn't work

### What happened?

When I try to run this code with data from https://play.reduct.store:

```js
for await (const record of bucket.query(entryName)) {
    console.info('read');
    const blob = await record.read();
}
```

The `bucket.query` method throws the following error:

```
[ERROR] TypeError [ERR_INVALID_ARG_TYPE]: The "iterable" argument must be an instance of Iterable. Received null
```

### Version

v1.7.1

### OS

Linux

### Snippet to repeat the bug

_No response_

### Logs

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [1.7.2] - 2023-10-28
+
+### Fixed:
+
+- RS-56: Queering batched records, [PR-71](https://github.com/reductstore/reduct-js/pull/71)
+
 ## [1.7.1] - 2023-10-11
 
 ### Fixed:

--- a/src/Bucket.ts
+++ b/src/Bucket.ts
@@ -335,7 +335,24 @@ export class Bucket {
                 }
                 stream = data;
             } else {
-                stream = Readable.from(head ? Buffer.from([]) : data.read(Number(size)));
+                let buffer = Buffer.from([]);
+                if (!head) {
+                    buffer = await new Promise((resolve, reject) => {
+                        data.on("readable", function handler() {
+                            const chunk = data.read(Number(size));
+                            if (chunk !== null) {
+                                resolve(chunk);
+                                data.off("readable", handler);
+                            }
+                        });
+
+                        data.on("error", (err: any) => {
+                            reject(err);
+                        });
+                    });
+
+                }
+                stream = Readable.from(buffer);
             }
 
             yield new ReadableRecord(BigInt(ts), size, last, stream, labels, contentType);

--- a/test/Bucket.test.ts
+++ b/test/Bucket.test.ts
@@ -153,6 +153,24 @@ describe("Bucket", () => {
         expect(md5(actual)).toEqual(md5(bigBlob));
     });
 
+    it("should query batched big blobs", async () => {
+        const bigBlob1 = crypto.randomBytes(2 ** 20);
+        const bigBlob2 = crypto.randomBytes(2 ** 20);
+
+        const bucket: Bucket = await client.getBucket("bucket");
+        let record = await bucket.beginWrite("big-blob");
+        await record.write(bigBlob1);
+
+        record = await bucket.beginWrite("big-blob");
+        await record.write(bigBlob2);
+
+        const blobs = await all(bucket.query("big-blob"));
+
+        expect(blobs.length).toEqual(2);
+        expect(md5(await blobs[0].read())).toEqual(md5(bigBlob1));
+        expect(md5(await blobs[1].read())).toEqual(md5(bigBlob2));
+    });
+
     it("should read write and read labels along with records", async () => {
         const bucket: Bucket = await client.getBucket("bucket");
         const record = await bucket.beginWrite("entry-1", {labels: {label1: "label1", label2: 100n, label3: true}});
@@ -238,7 +256,7 @@ describe("Bucket", () => {
         await expect(bucket.beginRead("entry-1")).rejects.toMatchObject({status: 404});
     });
 
-    it_api("1.7")("should write a batch of records", async () => {
+    it("should write a batch of records", async () => {
         const bucket: Bucket = await client.getBucket("bucket");
         const batch = await bucket.beginWriteBatch("entry-10");
         batch.add(1000n, "somedata1");

